src/main/java/com/strandls/document/service/Impl/DocumentServiceImpl.java

```diff
@@ -442,14 +465,29 @@ public MailData generateMailData(Long documentId) {
 
 	@Override
 	public Boolean removeDocument(HttpServletRequest request, Long documentId) {
-		CommonProfile profile = AuthUtil.getProfileFromRequest(request);
-		Long userId = Long.parseLong(profile.getId());
-		JSONArray userRoles = (JSONArray) profile.getAttribute("roles");
-		Document document = documentDao.findById(documentId);
-		if (document.getAuthorId().equals(userId) || userRoles.contains("ROLE_ADMIN")) {
-			document.setIsDeleted(true);
-			documentDao.update(document);
-			return true;
+		try {
+			CommonProfile profile = AuthUtil.getProfileFromRequest(request);
+			Long userId = Long.parseLong(profile.getId());
+			JSONArray userRoles = (JSONArray) profile.getAttribute("roles");
+			Document document = documentDao.findById(documentId);
+
+			MapQueryResponse esResponse;
+
+			esResponse = esService.delete(DocumentIndex.index.getValue(), DocumentIndex.type.getValue(),
+					documentId.toString());
+
+			ResultEnum result = esResponse.getResult();
+
+			if (result.getValue().equals("DELETED")
+					|| (document.getAuthorId().equals(userId) || userRoles.contains("ROLE_ADMIN"))) {
+				document.setIsDeleted(true);
+				documentDao.update(document);
+				System.out.print("===deleted docuemnt===" + documentId + "returned" + result.getValue());
+				return true;
+			}
+
+		} catch (ApiException e) {
+			e.printStackTrace();
```

Comment:
dont commit printstackTrace, use it only for development, use a logger and generic exception to handle the other exception as well